internal/http.go

```diff
@@ -48,6 +49,50 @@ func (h *basicAuthHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request)
 	h.next.ServeHTTP(rw, req)
 }
 
+type TokenAuthErrorFunc func(rw http.ResponseWriter)
+
+// TokenAuthHandler returns a http handler that requires `Authorization: Token <token>`
+// Introduced to support InfluxDB 2.x style authentication
+// https://v2.docs.influxdata.com/v2.0/reference/api/#authentication
+func TokenAuthHandler(token string, onError TokenAuthErrorFunc) func(h http.Handler) http.Handler {
```

Comment:
Set this up to handle a direct comparison with any Authorization header value, not treating the scheme and credentials separately, just as a single big string.  I think this will be more reusable and less tied to the InfluxDBv2 plugin.